src/Cli/dotnet/commands/dotnet-workload/history/WorkloadHistoryCommand.cs

```diff
@@ -0,0 +1,91 @@
+using Microsoft.NET.Sdk.WorkloadManifestReader;
+using Microsoft.DotNet.Configurer;
+using System.IO;
+using Microsoft.Deployment.DotNet.Releases;
+using Microsoft.DotNet.Cli.commands.dotnet_workload;
+
+
+namespace Microsoft.DotNet.Workloads.Workload.History
+{
+    internal class WorkloadHistoryCommand : WorkloadCommandBase
+    {
+        private readonly IInstaller _workloadInstaller;
+        private IWorkloadResolver _workloadResolver;
+        private readonly ReleaseVersion _sdkVersion;
+        private readonly string _dotnetPath;
+
+        public WorkloadHistoryCommand(
+            ParseResult parseResult,
+            IReporter reporter = null,
+            IWorkloadResolver workloadResolver = null,
+            IInstaller workloadInstaller = null,
+            INuGetPackageDownloader nugetPackageDownloader = null,
+            string dotnetDir = null,
+            string tempDirPath = null,
+            string version = null,
+            string userProfileDir = null
+        ) : base(parseResult, CommonOptions.HiddenVerbosityOption, reporter, tempDirPath, nugetPackageDownloader)
+        {
+
+            var creationParameters = new WorkloadResolverFactory.CreationParameters()
+            {
+                DotnetPath = dotnetDir,
+                UserProfileDir = userProfileDir,
+                GlobalJsonStartDir = null,
+                SdkVersionFromOption = parseResult.SafelyGetValueForOption(InstallingWorkloadCommandParser.VersionOption),
+                VersionForTesting = version,
+                CheckIfFeatureBandManifestExists = !(parseResult.SafelyGetValueForOption(InstallingWorkloadCommandParser.PrintDownloadLinkOnlyOption)),
+                WorkloadResolverForTesting = workloadResolver,
+                UseInstalledSdkVersionForResolver = true
+            };
+
+            var creationResult = WorkloadResolverFactory.Create(creationParameters);
+
+            _dotnetPath = creationResult.DotnetPath;
+            userProfileDir = creationResult.UserProfileDir;
+            _sdkVersion = creationResult.SdkVersion;
+            _workloadResolver = creationResult.WorkloadResolver;
+            var sdkFeatureBand = new SdkFeatureBand(_sdkVersion);
+
+            _workloadInstaller = workloadInstaller ??
+                                 WorkloadInstallerFactory.GetWorkloadInstaller(Reporter, sdkFeatureBand,
+                                     _workloadResolver, Verbosity, userProfileDir, VerifySignatures, PackageDownloader, dotnetDir, TempDirectoryPath,
+                                     packageSourceLocation: null, _parseResult.ToRestoreActionConfig());
+        }
+
+        public override int Execute()
+        {
+            var displayRecords = WorkloadHistoryDisplay.ProcessWorkloadHistoryRecords(_workloadInstaller.GetWorkloadHistoryRecords());
+
+            if (!displayRecords.Any())
+            {
+                //  TODO: Localize
+                Reporter.WriteLine("No workload history found");
```

Comment:
Should we handle creating the resources for each of these TODOs before merging?

Reply:
Ah, yeah; forgot about this. I should add it to my todo list 👍